File: db/fetch_price.py
import requests
import time
import json
from datetime import datetime

with open('db/tokens.json', 'r') as file:
    data = json.load(file)

tokens = [[item['id'], item['address']] for item in data]

while True:
    with open('db/prices.json', 'r') as file:
        
        try:
            data = json.load(file)
        except json.JSONDecodeError:
            data = []

        for token in tokens:
            
            uri = "https://api.dexscreener.com/latest/dex/tokens/"+ token[1]
            headers = {}
            response = requests.get(
                uri, headers=headers
            )

            if response.status_code == 200:
                response = response.json()
                
            price = response['pairs'][0]['priceUsd']
            
            token_id = str(time.time()).replace('.','')
            
            new_price = {
                "id": int(token_id),
                "token_id": token[0],
                "date": datetime.now().isoformat(),
                "price": price
            }
            
            print(data)
            data.append(new_price)
            print(data)

            with open('db/prices.json', 'w') as output:
                json.dump(data, output, indent=4)
            time.sleep(60)

File: db/fetch_price.php
<?php

function getTokens() {
    $file = 'db/tokens.json';
    if (!file_exists($file)) {
        return [];
    }
    
    $data = json_decode(file_get_contents($file), true);
    return array_map(fn($item) => [$item['id'], $item['address']], $data);
}

function getPrices() {
    $file = 'db/prices.json';
    if (!file_exists($file)) {
        return [];
    }
    
    $content = file_get_contents($file);
    return json_decode($content, true) ?: [];
}

function fetchPrice($tokenAddress) {
    $url = "https://api.dexscreener.com/latest/dex/tokens/" . $tokenAddress;
    $response = file_get_contents($url);
    
    if ($response === false) {
        return null;
    }
    
    $data = json_decode($response, true);
    return $data['pairs'][0]['priceUsd'] ?? null;
}

function savePrices($data) {
    file_put_contents('db/prices.json', json_encode($data, JSON_PRETTY_PRINT));
}

$tokens = getTokens();

while (true) {
    $data = getPrices();
    
    foreach ($tokens as $token) {
        $price = fetchPrice($token[1]);
        
        if ($price !== null) {
            $tokenId = str_replace('.', '', microtime(true));
            $newPrice = [
                "id" => (int) $tokenId,
                "token_id" => $token[0],
                "date" => date('c'),
                "price" => $price
            ];
            
            echo json_encode($newPrice, JSON_PRETTY_PRINT) . "\n";
            
            $data[] = $newPrice;
            savePrices($data);
        }
        
    }
    sleep(60);
}
